pyxem/components/scattering_fit_component.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+"""
+A HyperSpy fitting component to fit the independent atomic scattering
+background to a radial profile.
+
+"""
+import numpy as np
+
+from hyperspy.component import Component
+from pyxem.utils.atomic_scattering_params import ATOMIC_SCATTERING_PARAMS
+from pyxem.utils.lobato_scattering_params import ATOMIC_SCATTERING_PARAMS_LOBATO
+
+
+class ScatteringFitComponent(Component):
```

Comment:
Could this be re-written to inherit from the hyperspy Expression component rather than the base Component class?  It may not be the right choice in this case, but we should consider it. See: https://github.com/hyperspy/hyperspy/issues/2251

Reply:
Yes, but changes will be postponed until the issue with HyperSpy as_signal() (https://github.com/hyperspy/hyperspy/issues/2255) is resolved.